### ISSUE TITLE
Support `--no-status-rc` with `--help` and `--version`

### ISSUE THREAD

**Issue by pniederlag** [robotframework/robotframework#3874]
```
robot --version
echo $?
251
```

Is there a reason why robot reports an exit status of 251 von the --version command?
Any chance to turn this into a 0?

Thx for a great tool,
Peter

**Comment by pekkaklarck** [comment]
Return code zeor has been reserved for situation where tests were executed and no test failed. What use case you have for `--version` returning zero?

**Comment by pniederlag** [comment]
Hi,

thx for getting back on this.

I like to use ```--version```  commands inside build-pipelines to gather version-output from the used tools. Also I like to use it when starting to build pipelines in order to verify proper setup.

We had some troubles integration robot into our pipelines until we noticed they where breaking/failing due to the ```robot --version``` step exiting with non-zero and thus stoping the pipeline.

Once we know this fact we can of course work with it. It just feels a little strange that ```robot --version``` does exit with non-zero. AFAIR most other tools (npm, gradle, docker, jfrog, composer, ...) return 0 zero exit code on running the version command.

**Comment by pniederlag** [comment]
```
npm -version > /dev/null; echo $?; docker version > /dev/null; echo $?; gradle -version > /dev/null; echo $?; java -version 2&> /dev/null; echo $?; robot --version > /dev/null;echo $?
0
0
0
0
251
```

;)

**Comment by pniederlag** [comment]
So robot behaves different from a bunch of other tools. I think it would be nice if robot would follow the other tools here.

**Comment by perlun** [comment]
Yeah, I agree that this feels a bit non-standard to be honest. Is this something you could consider revisiting @pekkaklarck? :thinking: 

If we put it like this: what would be the _problem_ by returning zero both in this case and all tests executed successfully? For the record, `rspec` (a unit test framework popular in the Ruby community) also returns 0 on `--version`. Just like when unit tests have been executed successfully.

```
$ rspec --version ; echo $?
RSpec 3.9
  - rspec-core 3.9.1
  - rspec-expectations 3.9.1
  - rspec-mocks 3.9.1
  - rspec-support 3.9.2
0
```

**Comment by pekkaklarck** [comment]
I guess it would be better to return 0 than 251 when `--version` or `--help` is used. We should then make it an error to use these options with anything else, though. At the moment you can run, for example,

    robot --version tests.robot

and get version printed out without tests being run. In situations where users can pass command line options configuring execution, someone accidentally (or on purpose) passing `--version` shouldn't result with execution returning zero.

I don't consider this too high priority issue but if someone is interested to provide a PR this can be changed. Our command line option parsing code should be rewritten at some point, hopefully sooner than later, and this could be changed as part of that.

**Comment by ericbjones** [comment]
:(  
hoped `--nostatusrc` would at least work in a pipeline but it does not apply to `--version` output
```
❯ robot --version --nostatusrc; \
echo $?
Robot Framework 5.0.1 (Python 3.9.16 on darwin)
251
```
however wrapping in echo works fine in a pipeline and is honestly more descriptive of why you would run it in a pipeline anyways, so I agree not a high priority.  Just odd behavior that a successful cli command results in an error status.
```
❯ echo $(robot --version); \
echo $?
Robot Framework 5.0.1 (Python 3.9.16 on darwin)
0
```

**Comment by pekkaklarck** [comment]
`--version` and `--help` honoring `--nostatusrc` would be a good idea. Unlike changing the default behavior, it would be totally backwards compatible. It ought to be easy to implement as well. Should we do that?

**Comment by aaltat** [comment]
I think it would be good idea to make `--version` and `--help` honoring `--nostatusrc`

**Comment by pekkaklarck** [comment]
In RF 7.4 `--help` and `--version` will honor the `--no-status-rc` option. I get this now locally:
```
$ robot --version --no-status-rc
Robot Framework 7.4.dev1 (Python 3.8.10 on linux)
$ echo $?
0
```
The default exit code won't change. I consider it important that, by default, Robot returns 0 only if tests/tasks were executed and none failed. The change would also be backwards incompatible requiring a deprecation period. Safer and easier this way.
